Index: rustc/compiler/rustc_codegen_ssa/Cargo.toml
===================================================================
--- rustc.orig/compiler/rustc_codegen_ssa/Cargo.toml
+++ rustc/compiler/rustc_codegen_ssa/Cargo.toml
@@ -49,7 +49,3 @@ libc = "0.2.50"
 version = "0.31.1"
 default-features = false
 features = ["read_core", "elf", "macho", "pe", "xcoff", "unaligned", "archive", "write"]
-
-[target.'cfg(windows)'.dependencies.windows]
-version = "0.48.0"
-features = ["Win32_Globalization"]
Index: rustc/compiler/rustc_data_structures/Cargo.toml
===================================================================
--- rustc.orig/compiler/rustc_data_structures/Cargo.toml
+++ rustc/compiler/rustc_data_structures/Cargo.toml
@@ -37,16 +37,6 @@ itertools = "0.10.1"
 [dependencies.parking_lot]
 version = "0.11"
 
-[target.'cfg(windows)'.dependencies.windows]
-version = "0.48.0"
-features = [
-    "Win32_Foundation",
-    "Win32_Storage_FileSystem",
-    "Win32_System_IO",
-    "Win32_System_ProcessStatus",
-    "Win32_System_Threading",
-]
-
 [target.'cfg(not(target_arch = "wasm32"))'.dependencies]
 memmap2 = "0.2.1"
 
Index: rustc/compiler/rustc_driver_impl/Cargo.toml
===================================================================
--- rustc.orig/compiler/rustc_driver_impl/Cargo.toml
+++ rustc/compiler/rustc_driver_impl/Cargo.toml
@@ -56,12 +56,6 @@ rustc_mir_transform = { path = "../rustc
 [target.'cfg(unix)'.dependencies]
 libc = "0.2"
 
-[target.'cfg(windows)'.dependencies.windows]
-version = "0.48.0"
-features = [
-    "Win32_System_Diagnostics_Debug",
-]
-
 [features]
 llvm = ['rustc_interface/llvm']
 max_level_info = ['rustc_log/max_level_info']
Index: rustc/compiler/rustc_errors/Cargo.toml
===================================================================
--- rustc.orig/compiler/rustc_errors/Cargo.toml
+++ rustc/compiler/rustc_errors/Cargo.toml
@@ -26,13 +26,5 @@ termize = "0.1.1"
 serde = { version = "1.0.125", features = [ "derive" ] }
 serde_json = "1.0.59"
 
-[target.'cfg(windows)'.dependencies.windows]
-version = "0.48.0"
-features = [
-    "Win32_Foundation",
-    "Win32_Security",
-    "Win32_System_Threading",
-]
-
 [features]
 rustc_use_parallel_compiler = ['rustc_error_messages/rustc_use_parallel_compiler']
Index: rustc/compiler/rustc_session/Cargo.toml
===================================================================
--- rustc.orig/compiler/rustc_session/Cargo.toml
+++ rustc/compiler/rustc_session/Cargo.toml
@@ -25,10 +25,3 @@ termize = "0.1.1"
 
 [target.'cfg(unix)'.dependencies]
 libc = "0.2"
-
-[target.'cfg(windows)'.dependencies.windows]
-version = "0.48.0"
-features = [
-    "Win32_Foundation",
-    "Win32_System_LibraryLoader",
-]
Index: rustc/src/bootstrap/Cargo.toml
===================================================================
--- rustc.orig/src/bootstrap/Cargo.toml
+++ rustc/src/bootstrap/Cargo.toml
@@ -62,21 +62,6 @@ semver = "1.0.17"
 [target.'cfg(not(target_os = "solaris"))'.dependencies]
 fd-lock = "3.0.8"
 
-[target.'cfg(windows)'.dependencies.junction]
-version = "1.0.0"
-
-[target.'cfg(windows)'.dependencies.windows]
-version = "0.46.0"
-features = [
-    "Win32_Foundation",
-    "Win32_Security",
-    "Win32_System_Diagnostics_Debug",
-    "Win32_System_JobObjects",
-    "Win32_System_ProcessStatus",
-    "Win32_System_Threading",
-    "Win32_System_Time",
-]
-
 [dev-dependencies]
 pretty_assertions = "1.2"
 
Index: rustc/src/tools/compiletest/Cargo.toml
===================================================================
--- rustc.orig/src/tools/compiletest/Cargo.toml
+++ rustc/src/tools/compiletest/Cargo.toml
@@ -27,13 +27,3 @@ anyhow = "1"
 
 [target.'cfg(unix)'.dependencies]
 libc = "0.2"
-
-[target.'cfg(windows)'.dependencies]
-miow = "0.5"
-
-[target.'cfg(windows)'.dependencies.windows]
-version = "0.48.0"
-features = [
-    "Win32_Foundation",
-    "Win32_System_Diagnostics_Debug",
-]
Index: rustc/src/tools/rust-analyzer/crates/stdx/Cargo.toml
===================================================================
--- rustc.orig/src/tools/rust-analyzer/crates/stdx/Cargo.toml
+++ rustc/src/tools/rust-analyzer/crates/stdx/Cargo.toml
@@ -19,10 +19,6 @@ jod-thread = "0.1.2"
 crossbeam-channel = "0.5.5"
 # Think twice before adding anything here
 
-[target.'cfg(windows)'.dependencies]
-miow = "0.5.0"
-winapi = { version = "0.3.9", features = ["winerror"] }
-
 [features]
 # Uncomment to enable for the whole crate graph
 # default = [ "backtrace" ]
Index: rustc/src/tools/rust-analyzer/crates/rust-analyzer/Cargo.toml
===================================================================
--- rustc.orig/src/tools/rust-analyzer/crates/rust-analyzer/Cargo.toml
+++ rustc/src/tools/rust-analyzer/crates/rust-analyzer/Cargo.toml
@@ -77,12 +77,6 @@ tt.workspace = true
 vfs-notify.workspace = true
 vfs.workspace = true
 
-[target.'cfg(windows)'.dependencies]
-winapi = "0.3.9"
-
-[target.'cfg(not(target_env = "msvc"))'.dependencies]
-jemallocator = { version = "0.5.0", package = "tikv-jemallocator", optional = true }
-
 [dev-dependencies]
 expect-test = "1.4.0"
 xshell = "0.2.2"
Index: rustc/src/tools/cargo/Cargo.toml
===================================================================
--- rustc.orig/src/tools/cargo/Cargo.toml
+++ rustc/src/tools/cargo/Cargo.toml
@@ -53,7 +53,6 @@ libc = "0.2.144"
 libgit2-sys = "0.15.1"
 log = "0.4.17"
 memchr = "2.1.3"
-miow = "0.5.0"
 opener = "0.5"
 openssl ="0.10.55"
 os_info = "3.5.0"
@@ -91,7 +90,6 @@ unicode-xid = "0.2.0"
 url = "2.2.2"
 varisat = "0.2.1"
 walkdir = "2.3.1"
-windows-sys = "0.48"
 
 [package]
 name = "cargo"
@@ -174,19 +172,6 @@ walkdir.workspace = true
 [target.'cfg(not(windows))'.dependencies]
 openssl = { workspace = true, optional = true }
 
-[target.'cfg(windows)'.dependencies]
-fwdansi.workspace = true
-
-[target.'cfg(windows)'.dependencies.windows-sys]
-workspace = true
-features = [
-  "Win32_Foundation",
-  "Win32_Storage_FileSystem",
-  "Win32_System_Console",
-  "Win32_System_Threading",
-  "Win32_System_JobObjects",
-]
-
 [dev-dependencies]
 cargo-test-macro.workspace = true
 cargo-test-support.workspace = true
Index: rustc/src/tools/cargo/crates/cargo-util/Cargo.toml
===================================================================
--- rustc.orig/src/tools/cargo/crates/cargo-util/Cargo.toml
+++ rustc/src/tools/cargo/crates/cargo-util/Cargo.toml
@@ -22,7 +22,3 @@ walkdir.workspace = true
 
 [target.'cfg(target_os = "macos")'.dependencies]
 core-foundation.workspace = true
-
-[target.'cfg(windows)'.dependencies]
-miow.workspace = true
-windows-sys = { workspace = true, features = ["Win32_Storage_FileSystem", "Win32_Foundation", "Win32_System_Console"] }
Index: rustc/src/tools/cargo/crates/cargo-test-support/Cargo.toml
===================================================================
--- rustc.orig/src/tools/cargo/crates/cargo-test-support/Cargo.toml
+++ rustc/src/tools/cargo/crates/cargo-test-support/Cargo.toml
@@ -28,6 +28,3 @@ termcolor.workspace = true
 time.workspace = true
 toml.workspace = true
 url.workspace = true
-
-[target.'cfg(windows)'.dependencies]
-windows-sys = { workspace = true, features = ["Win32_Storage_FileSystem"] }
Index: rustc/src/tools/cargo/crates/home/Cargo.toml
===================================================================
--- rustc.orig/src/tools/cargo/crates/home/Cargo.toml
+++ rustc/src/tools/cargo/crates/home/Cargo.toml
@@ -15,6 +15,3 @@ license = "MIT OR Apache-2.0"
 readme = "README.md"
 repository = "https://github.com/rust-lang/cargo"
 description = "Shared definitions of home directories."
-
-[target.'cfg(windows)'.dependencies]
-windows-sys = { workspace = true, features = ["Win32_Foundation", "Win32_UI_Shell"] }
