diff --git a/src/tools/cargo/Cargo.toml b/src/tools/cargo/Cargo.toml
index 987b0e3dee..2423d89ddb 100644
--- a/src/tools/cargo/Cargo.toml
+++ b/src/tools/cargo/Cargo.toml
@@ -122,6 +122,5 @@ doc = false
 
 [features]
 deny-warnings = []
-vendored-openssl = ["openssl/vendored"]
-vendored-libgit2 = ["libgit2-sys/vendored"]
+# Debian: removed vendoring flags
 pretty-env-logger = ["pretty_env_logger"]
