--- a/src/bootstrap/builder.rs
+++ b/src/bootstrap/builder.rs
@@ -159,7 +159,7 @@
                 pathset, self.name, builder.config.exclude
             );
         }
-        let hosts = &builder.hosts;
+        //let hosts = &builder.hosts;
 
         // Determine the targets participating in this rule.
         let targets = if self.only_hosts {
@@ -172,17 +172,17 @@
             &builder.targets
         };
 
-        for host in hosts {
+        //for host in hosts {
             for target in targets {
                 let run = RunConfig {
                     builder,
                     path: pathset.path(builder),
-                    host: *host,
+                    host: builder.build.build,
                     target: *target,
                 };
                 (self.make_run)(run);
             }
-        }
+        //}
     }
 
     fn run(v: &[StepDescription], builder: &Builder<'_>, paths: &[PathBuf]) {
--- a/config.toml.example
+++ b/config.toml.example
@@ -115,12 +115,11 @@
 
 # In addition to the build triple, other triples to produce full compiler
 # toolchains for. Each of these triples will be bootstrapped from the build
-# triple and then will continue to bootstrap themselves. This platform must
-# currently be able to run all of the triples provided here.
+# triple and then will continue to bootstrap themselves.
 #host = ["x86_64-unknown-linux-gnu"]   # defaults to just the build triple
 
 # In addition to all host triples, other triples to produce the standard library
-# for. Each host triple will be used to produce a copy of the standard library
+# for. Each build triple will be used to produce a copy of the standard library
 # for each target triple.
 #target = ["x86_64-unknown-linux-gnu"] # defaults to just the build triple
 
