Index: rustc.git/src/bootstrap/lib.rs
===================================================================
--- rustc.git.orig/src/bootstrap/lib.rs
+++ rustc.git/src/bootstrap/lib.rs
@@ -777,10 +777,9 @@ impl Build {
 
         match which {
             GitRepo::Rustc => {
-                let sha = self.rust_sha().unwrap_or(channel::CFG_RELEASE_NUM);
-                Some(format!("/rustc/{}", sha))
+                Some(format!("/usr/src/rustc-{}", channel::CFG_RELEASE_NUM))
             }
-            GitRepo::Llvm => Some(String::from("/rustc/llvm")),
+            GitRepo::Llvm => panic!("GitRepo::Llvm unsupported on Debian"),
         }
     }
 
