--- a/src/bootstrap/doc.rs
+++ b/src/bootstrap/doc.rs
@@ -364,6 +364,8 @@
                 .arg(&builder.src.join("src/doc/index.md"))
                 .arg("--markdown-playground-url")
                 .arg("https://play.rust-lang.org/")
+                .arg("--markdown-css")
+                .arg("rust.css")
                 .arg("-o")
                 .arg(&out)
                 .arg(&path);
@@ -371,12 +373,6 @@
             if !builder.config.docs_minification {
                 cmd.arg("--disable-minification");
             }
-
-            if filename == "not_found.md" {
-                cmd.arg("--markdown-css").arg("https://doc.rust-lang.org/rust.css");
-            } else {
-                cmd.arg("--markdown-css").arg("rust.css");
-            }
             builder.run(&mut cmd);
         }
 
