Bug: https://github.com/rust-lang/rust/issues/68973s

Index: rustc.git/src/librustc_target/spec/windows_gnu_base.rs
===================================================================
--- rustc.git.orig/src/librustc_target/spec/windows_gnu_base.rs
+++ rustc.git/src/librustc_target/spec/windows_gnu_base.rs
@@ -33,6 +33,8 @@ pub fn opts() -> TargetOptions {
         "-lmsvcrt".to_string(),
         "-luser32".to_string(),
         "-lkernel32".to_string(),
+        "-lssp_nonshared".to_string(),
+        "-lssp".to_string(),
     ];
     late_link_args.insert(LinkerFlavor::Gcc, mingw_libs.clone());
     late_link_args.insert(LinkerFlavor::Lld(LldFlavor::Ld), mingw_libs);
