Bug: https://github.com/rust-lang/rust/issues/78932
--- a/src/bootstrap/dist.rs
+++ b/src/bootstrap/dist.rs
@@ -2370,7 +2370,8 @@
 /// Note: This function does not yet support Windows, but we also don't support
 ///       linking LLVM tools dynamically on Windows yet.
 fn maybe_install_llvm(builder: &Builder<'_>, target: TargetSelection, dst_libdir: &Path) {
-    if !builder.config.llvm_link_shared {
+    // Debian does not need to install this
+    if true {
         // We do not need to copy LLVM files into the sysroot if it is not
         // dynamically linked; it is already included into librustc_llvm
         // statically.
