Bug: https://github.com/rust-lang/rust/issues/84038
--- a/src/test/ui/asm/naked-invalid-attr.rs
+++ b/src/test/ui/asm/naked-invalid-attr.rs
@@ -1,6 +1,10 @@
 // Checks that #[naked] attribute can be placed on function definitions only.
 //
 // ignore-wasm32 asm unsupported
+// ignore-powerpc64 asm unsupported
+// ignore-powerpc64le asm unsupported
+// ignore-s390x asm unsupported
+// ignore-sparc64 asm unsupported
 #![feature(asm)]
 #![feature(naked_functions)]
 #![naked] //~ ERROR should be applied to a function definition
--- a/src/test/ui/feature-gates/feature-gate-naked_functions.rs
+++ b/src/test/ui/feature-gates/feature-gate-naked_functions.rs
@@ -1,3 +1,7 @@
+// ignore-powerpc64 asm unsupported
+// ignore-powerpc64le asm unsupported
+// ignore-s390x asm unsupported
+// ignore-sparc64 asm unsupported
 #![feature(asm)]
 
 #[naked]
--- a/src/test/ui/rfc-2091-track-caller/error-with-naked.rs
+++ b/src/test/ui/rfc-2091-track-caller/error-with-naked.rs
@@ -1,3 +1,7 @@
+// ignore-powerpc64 asm unsupported
+// ignore-powerpc64le asm unsupported
+// ignore-s390x asm unsupported
+// ignore-sparc64 asm unsupported
 #![feature(asm, naked_functions)]
 
 #[track_caller] //~ ERROR cannot use `#[track_caller]` with `#[naked]`
