Bug: https://github.com/rust-lang/rust/issues/79009

--- a/src/test/debuginfo/extern-c-fn.rs
+++ b/src/test/debuginfo/extern-c-fn.rs
@@ -1,3 +1,4 @@
+// ignore-test fails on gdb-10 see #79009
 // min-lldb-version: 310
 
 // compile-flags:-g
--- a/src/test/debuginfo/generator-objects.rs
+++ b/src/test/debuginfo/generator-objects.rs
@@ -1,3 +1,4 @@
+// ignore-test fails on gdb-10 see #79009
 // ignore-tidy-linelength
 
 // Require a gdb that can read DW_TAG_variant_part.
--- a/src/test/debuginfo/issue-57822.rs
+++ b/src/test/debuginfo/issue-57822.rs
@@ -1,3 +1,4 @@
+// ignore-test fails on gdb-10 see #79009
 // This test makes sure that the LLDB pretty printer does not throw an exception
 // for nested closures and generators.
 
--- a/src/test/debuginfo/pretty-huge-vec.rs
+++ b/src/test/debuginfo/pretty-huge-vec.rs
@@ -1,3 +1,4 @@
+// ignore-test fails on gdb-10 see #79009
 // ignore-windows failing on win32 bot
 // ignore-freebsd: gdb package too new
 // ignore-android: FIXME(#10381)
