Description: Various fixes for Rust on MIPS
Author: Dragan Mladjenovic <dmladjenovic@wavecomp.com>
Bug-Debian: https://bugs.debian.org/881845
Last-Update: 2020-03-23

--- a/compiler/rustc_codegen_llvm/src/llvm_util.rs
+++ b/compiler/rustc_codegen_llvm/src/llvm_util.rs
@@ -113,6 +113,12 @@
         // Use non-zero `import-instr-limit` multiplier for cold callsites.
         add("-import-cold-multiplier=0.1", false);
 
+        if sess.target.arch == "mips"
+            || sess.target.arch == "mips64"
+        {
+            add("-fast-isel=0", false);
+        }
+
         for arg in sess_args {
             add(&(*arg), true);
         }
