--- a/src/librustc_llvm/build.rs
+++ b/src/librustc_llvm/build.rs
@@ -229,7 +229,7 @@
         if lib.starts_with("-LIBPATH:") {
             println!("cargo:rustc-link-search=native={}", &lib[9..]);
         } else if is_crossed {
-            if lib.starts_with("-L") {
+            if lib.starts_with("-L") && lib[2..].contains(&host) {
                 println!("cargo:rustc-link-search=native={}",
                          lib[2..].replace(&host, &target));
             }
