Index: rustc.git/compiler/rustc_llvm/build.rs
===================================================================
--- rustc.git.orig/compiler/rustc_llvm/build.rs
+++ rustc.git/compiler/rustc_llvm/build.rs
@@ -249,7 +249,7 @@ fn main() {
         if is_crossed {
             if lib.starts_with("-LIBPATH:") {
                 println!("cargo:rustc-link-search=native={}", lib[9..].replace(&host, &target));
-            } else if lib.starts_with("-L") {
+            } else if lib.starts_with("-L") && lib[2..].contains(&host) {
                 println!("cargo:rustc-link-search=native={}", lib[2..].replace(&host, &target));
             }
         } else if lib.starts_with("-LIBPATH:") {
