Index: rustc/library/profiler_builtins/build.rs
===================================================================
--- rustc.orig/library/profiler_builtins/build.rs
+++ rustc/library/profiler_builtins/build.rs
@@ -13,6 +13,18 @@ fn main() {
     }
 
     let target = env::var("TARGET").expect("TARGET was not set");
+    if target == std::env::var("HOST").unwrap() {
+        // use system profiler runtime if possible
+        println!(
+            "cargo:rustc-link-search=/usr/lib/clang/{}/lib/linux/",
+            env::var("DEB_LLVM_VERSION").unwrap()
+        );
+        println!(
+            "cargo:rustc-link-lib=static=clang_rt.profile-{}",
+            env::var("DEB_CLANG_RT_ARCH").unwrap()
+        );
+        return;
+    }
     let cfg = &mut cc::Build::new();
 
     // FIXME: `rerun-if-changed` directives are not currently emitted and the build script
